### DISCUSSION TITLE
classes/class_arraymesh

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#259] [User-contributed notes]
# classes/class_arraymesh

Inherits: Mesh< Resource< RefCounted< Object Mesh type that provides utility for constructing a surface from arrays. Description: The ArrayMesh is used to construct a Mesh by specifying the attribu...

https://docs.godotengine.org/en/stable/classes/class_arraymesh.html

**Comment by tx350z** [comment]
It should be mentioned that the maximum number of surfaces is 256.

**Reply by onze** [reply]
By the Mesh [description](https://docs.godotengine.org/en/stable/classes/class_mesh.html#description):
> The maximum number of surfaces per mesh is RenderingServer.MAX_MESH_SURFACES.
